tests/beaker_tests/lib/utilitylib.rb

```diff
@@ -291,6 +291,39 @@ def test_idempotence(tests, id)
   end
 end
 
+# Helper to retrieve all instances of a given resource
+# @param agent [String] the agent to be queried
+# @param res_name [String] the resource to retrieve instances of
+# @return [Array] an array of string names of instances
+def get_current_resource_instances(agent, res_name)
+  cmd_str = UtilityLib.get_namespace_cmd(agent, UtilityLib::PUPPET_BINPATH +
+      "resource #{res_name}", options)
+  on(agent, cmd_str, acceptable_exit_codes: [0])
+  names = stdout.scan(/#{res_name} { '(.+)':/).flatten
+  names
+end
+
+# Method to clean by putting a resource in absent state
+# @param agent [String] the agent that is going to run the test
+# @param res_name [String] the resource name that will be cleaned up
+def manifest_absent_cleanup(agent, res_name, stepinfo='absent clean')
+  # Don't forget to UtilityLib.set_manifest_path(master, self)
+  step "TestStep :: #{stepinfo}" do
+    # create an absent resource for each instance within node default
+    abs_str = ''
+    get_current_resource_instances(agent, res_name).each do |title|
+      abs_str << "#{res_name} { '#{title}':\n    ensure => absent,\n  }\n"
+    end
+    manifest_str = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}\n" \
```

Comment:
Instead of applying a manifest I think @chrisvanheuveln 's suggestion is better.

Mike and I had discussed using puppet resource <provider> to get current state and then follow it with puppet resource <provider> ensure=absent to cleanup.